pkg/api/protobuf/go/minder/v1/validators.go

```diff
@@ -232,14 +233,33 @@ func (jq *RuleType_Definition_Eval_JQComparison) Validate() error {
 		return fmt.Errorf("%w: jq ingested definition is empty", ErrInvalidRuleTypeDefinition)
 	}
 
+	q, err := gojq.Parse(jq.GetIngested().GetDef())
+	if err != nil {
+		return fmt.Errorf("%w: jq ingested definition is not parsable: %s", ErrInvalidRuleTypeDefinition, err)
+	}
+	if _, err = gojq.Compile(q); err != nil {
+		return fmt.Errorf("%w: jq ingested definition is invalid: %s", ErrInvalidRuleTypeDefinition, err)
+	}
+
 	if jq.GetProfile() != nil && jq.GetConstant() != nil {
 		return fmt.Errorf("%w: jq profile and constant accessors are mutually exclusive", ErrInvalidRuleTypeDefinition)
 	} else if jq.GetProfile() == nil && jq.GetConstant() == nil {
 		return fmt.Errorf("%w: jq missing profile or constant accessor", ErrInvalidRuleTypeDefinition)
 	}
 
-	if jq.GetProfile() != nil && jq.GetProfile().GetDef() == "" {
-		return fmt.Errorf("%w: jq profile accessor definition is empty", ErrInvalidRuleTypeDefinition)
+	if jq.GetProfile() != nil {
+		if jq.GetProfile().GetDef() == "" {
+			return fmt.Errorf("%w: jq profile accessor definition is empty", ErrInvalidRuleTypeDefinition)
+		}
+
+		q, err := gojq.Parse(jq.GetProfile().GetDef())
+		if err != nil {
+			return fmt.Errorf("%w: jq profile accessor definition is invalid: %s", ErrInvalidRuleTypeDefinition, err)
+		}
+		if _, err = gojq.Compile(q); err != nil {
+			return fmt.Errorf("%w: jq profile accessor definition is invalid: %s", ErrInvalidRuleTypeDefinition, err)
+		}
```

Comment:
Should we define a `Validate` for `RuleType_Definition_Eval_JQComparison_Operator`, and then this code could be:

```golang
if jq == nil {
  return fmt.Errorf(...)
}
// Some stuff that could be a CEL validation expression
if jq.GetProfile != nil && jq.GetConstant() != nil {
  fmt.Errorf("... mutually exclusive")
} else if jq.GetProfile() == nill && jq.GetConstant() == nil {
  fmt.Errorf("... missing either component")
}
if err := jq.GetIngested().Validate(); err != nil {
  fmt.Errorf("...", err)
}
if err := jq.GetProfile().Validate() err != nil {
  fmt.Errorf("...', err)
}
```

Reply:
Thanks for the suggestion, I've applied it